File: src/nes/cpu_test.go
package nes

import "testing"

func Test_0xa9_lda_immediate_load_data(t *testing.T) {
	cpu := NewCPU()
	if err := cpu.Run([]byte{0xa9, 0x05, 0x00}); err != nil {
		t.Fatal(err)
	}

	if cpu.registerA != 0x05 {
		t.Fatalf("")
	}
}

File: tests/cpu_test.cpp
#include "nes.h"
#include <gtest/gtest.h>

TEST(BasicInst, test_0xa9_lda_immediate_load_data) {
    Processor cpu;
    u8 insts[] = {0xa9, 0x05, 0x00};
    cpu.run(insts, 3);
    EXPECT_FALSE(cpu.status().is_zero());
    EXPECT_FALSE(cpu.status().is_negative());
}

TEST(BasicInst, test_0xa9_lda_zero_flag) {
    Processor cpu;
    u8 insts[] = {0xa9, 0x00, 0x00};
    cpu.run(insts, 3);
    EXPECT_TRUE(cpu.status().is_zero());
}
